### PR TITLE
fix server rendering

### LINKED ISSUE #18: server rendering
Hi !

I have an issue when pre-rendering pages from server when using `in-viewport`. classic access to unknow `global.document`. i fixed quickly here https://github.com/revolunet/in-viewport/commit/57e5db365aee600af91d7164313b7355ff9f456b tell me if you want a PR

Cheers and thanks for that nice lib :)

### PR DESCRIPTION
**Pull request by revolunet**
Just a quick workaround to allow server-side rendering

Should close #18

### CODE REVIEW

File: .zuul.yml [hunk 1]
```diff
@@ -5,17 +5,15 @@ scripts:
...
-  - name: safari
-    version: 5..latest
-  - name: iphone
-    version: 6.0..latest
-  - name: ipad
-    version: 6.0..latest
-  - name: android
-    version: 4.1..latest
+  # - name: firefox
+  #   version: 48..latest
```
Comment (tzi) [inline .zuul.yml]: We know there is some problem with the tests on iPad and iPhone.
But there is no reason to drop Firefox, IE, Safari and android :hamster:

File: in-viewport.js [hunk 2]
```diff
@@ -53,22 +53,27 @@ function debounce(func, wait, immediate) {
...
-        if (b === a) {
-          return true;
-        }
-      }
-      return false;
-    };
+var contains = function() {
+  if (global.document) {
+    return global.document.documentElement.compareDocumentPosition ?
+            function (a, b) {
```
Comment (tzi) [inline in-viewport.js]: You should keep the 2 space indentation :rabbit: 
Maybe we should add a `.editorconfig` file?

File: package.json [hunk 3]
```diff
@@ -21,7 +21,7 @@
     "deumdify": "^1.2.2",
     "merge": "1.2.0",
     "watchify": "^3.6.1",
-    "zuul": "^3.7.3",
-    "zuul-ngrok": "3.0.0"
+    "zuul": "^3.11.0",
+    "zuul-ngrok": "^4.0.0"
```
Comment (tzi) [inline package.json]: Thanks for the update :+1:

File: .zuul.yml (hunk 1 above)
Comment (vvo) [inline .zuul.yml]: HéHé same :)

**Comment by tzi** [comment]
@vvo We should probably remove the travis-ci, no?

**Comment by vvo** [comment]
> @vvo We should probably remove the travis-ci, no?

We can keep it but only test all latest versions + IE8 => IE 11. That should do it.

**Comment by vvo** [comment]
> We can keep it but only test all latest versions + IE8 => IE 11. That should do it.

This can be done in a separate PR, this is not the goal of that one. No big deal if tests here are not passing.

**Comment by tzi** [comment]
The problem with travis-ci is not the browser support, but the saucelabs credentials.
I think I never see it working here :laughing:

**Comment by vvo** [comment]
Oh yes yes, it's just because we encrypt the saucelabs KEY so it's only available to builds originating from vvo/in-viewport, not from forks. That's travis-ci security measure around encrypted env variables (understandable).

Once merged, or if a branch from this repo, tests will launch.

https://github.com/vvo/in-viewport/blob/master/.travis.yml

**Comment by revolunet** [comment]
thanks for the feedback, cleaned up :)

**Comment by revolunet** [comment]
thanks and happy week

**Comment by vvo** [comment]
Thanks for this :)

**Comment by tzi** [comment]
Thanks :smile:

**Pull request merged**